### PR TITLE
EWL-6547: Resource page - mobile. tables not showing complete data

### PR DESCRIPTION
**Pull request by russom-woldezghi**
## Ticket(s)
- [EWL-6547: Resource page - mobile. tables not showing complete data](https://issues.ama-assn.org/browse/EWL-6547)
- [EWL-6548: Resource page - mobile. Tables grey/white not displaying correctly](https://issues.ama-assn.org/browse/EWL-6548)

## Description
Adds height to `td` table cells for mobile displays

Depends on https://github.com/AmericanMedicalAssociation/ama-d8/pull/1033

## To Test
- [ ] Checkout branch to `EWL-6547-mobile-showing-table-header`
- [ ] Run `gulp serve`
- [ ] Run `drush @one.local cr` in Drupal VM
- [ ] Reduce browser to mobile viewport
- [ ] Visit http://ama-one.local/about-ama/cpt-editorial-panel/cptr-editorial-panel-meeting-process-calendar#tables (check column title appear as data-title attributes - for ones with values)
- [ ] Visit http://ama-one.local/member-groups-sections/international-medical-graduates/imgs-meeting-agenda-important-dates#schedules
- [ ] Observe the table alternating row colors go from grey to white consistently
- [ ] Did you test in IE 11?

## Visual Regressions

N/A

### Before
screen shot 2018-11-07 at 2 47 50 pm

### After
screen shot 2018-11-07 at 2 47 05 pm

## Remaining Tasks
Things relevant to here may be updating [The Glossary](https://issues.ama-assn.org:8446/confluence/display/DTD/Glossary?src=contextnavpagetreemode) with the latest naming changes or review of the relative [D8](https://github.com/AmericanMedicalAssociation/ama-d8) work.

## Additional Notes
Drupal PR this PR is dependent on: https://github.com/AmericanMedicalAssociation/ama-d8/pull/1033

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

**Comment by russom-woldezghi** [comment]
There is a `source/assets/js/tables.js` in the style guide that is doing something similar, but not addressing the inclusion of `data-title` attributes to `td` of table elements. I opted to approach this fix from Drupal as that seems to address the issue. Let me know if there should be additional work around the the `table.js` file to avoid duplication.

**Review by damontgomery — COMMENTED** [review]
Setting the height doesn't work for the example page.

overlapping

Is there a dynamic solution that we can use that works for all lengths of titles?

Maybe a header above the content?

stacked

```
tbody tr td:before {
    font-weight: bold;
    text-transform: uppercase;
    content: attr(data-title);
    display: block;
}

tbody tr td {
    border: 1px solid #EDEDED;
    display: block;
    padding: 2px;
    position: relative;
    -webkit-hyphens: auto;
    -ms-hyphens: auto;
    hyphens: auto;
}
```

The testing steps in this PR and the D8 PR are very helpful. The comments are also very helpful. Thanks.

**Comment by russom-woldezghi** [comment]
Hey @designerneil, for mobile display of tables, the two column layout in the styleguide: http://localhost:3000/?p=atoms-table might not be suitable for lengthy title/column header text in tables (see Dan's screenshot). Would Dan's suggest above need the styleguide, design assets such as Zeplin updated?

**Comment by russom-woldezghi** [comment]
@damontgomery - I have added the suggestion you posted above, plus some additional padding so its not so close to the edge of the tables on mobile. 
screen shot 2018-11-08 at 10 27 50 am

For the schedule section: http://ama-one.local/member-groups-sections/international-medical-graduates/imgs-meeting-agenda-important-dates#schedules the content doesn't seem to match what is on the previous example (in the description/testing of the PR). This could be a content issue?

**Comment by russom-woldezghi** [comment]
@designerneil Can you take a look at this issue?  
screen shot 2018-11-08 at 10 27 50 am

I applied your changes from the other PR that is closed #525.

**Review by designerneil — APPROVED** [review]

File: styleguide/source/assets/scss/01-atoms/_table.scss [hunk 1]
```diff
@@ -70,18 +70,20 @@ tbody {
...
       display: block;
-      padding: 2px 2px 2px 50%;
+      padding: 10px;
       position: relative;
+      -webkit-hyphens: auto;
+      -ms-hyphens: auto;
       hyphens: auto;
 
+      &.odd,
+      &:nth-child(odd)  {
```
Comment (Brian-Clement) [inline styleguide/source/assets/scss/01-atoms/_table.scss]: Changing this background from odd `tr` to odd `td` is creating the effect of column striping versus row striping.  Is this intentional?

**Review by Brian-Clement — COMMENTED** [review]
@russom-woldezghi question on the striping.  The `background: $gray-7;` has moved from `tr` to `td`, which is creating striped columns instead of striped rows.  Not sure if this is intended or not, here are the effects on different viewport sizes:

# Desktop

## Before
image


## After
image


# Tablet

## Before
image


## After
image


# Mobile

## Before
image


## After
image

**Comment by Brian-Clement** [comment]
One other thing I'm seeing here @russom-woldezghi is the format of the table on mobile.  The column headers in the Zeplin are to the left of the cells (zpl.io/295JwNp), but in this PR they are appearing above every cell.

File: styleguide/source/assets/scss/01-atoms/_table.scss (hunk 1 above)
Comment (russom-woldezghi) [reply styleguide/source/assets/scss/01-atoms/_table.scss]: I don't think the column striping in the desktop/tablet viewport is intentional. I can take a look. 

CC: @designerneil Do you have any thoughts on the background color for desktop/tablet viewport? 

The column stacking was decided to accommodate for lengthy title for mobile. It was a two-column before and text was getting cut off and overlapping. See below:

image

**Comment by russom-woldezghi** [comment]
@Brian-Clement Are you able to grant me access to the Zeplin account? I am not able to access the link. 

I have updated the stripping pattern to be a little more consistent across the desktop and tablet viewports. The update fixes the alternating rows, instead of columns for desktop and tablet. 

## Desktop
screen shot 2018-11-13 at 10 59 38 am

## Tablet
screen shot 2018-11-13 at 11 00 40 am

## Mobile
The two gray background, circled in red (see below) are cases when stripping by odd or even can produce unintended background color order for mobile viewport in a stacking like display. We chose to stack the cells to accommodate for length titles, cell values. The two column layout caused the text to be overlapped and harder to read the content on mobile viewport. 
 
### Before
48377341-2a5b2f00-e693-11e8-9f3a-ab57fe4a2169

### After
I also added a bottom-border to highlight the background color consistency of cell background color on different viewports. 

screen shot 2018-11-13 at 11 01 02 am

**Comment by damontgomery** [comment]
Since there are a few changes to tables that are overlapping, Neil is going to gather more information on the desired behavior and circle back.

**Pull request closed without merging**